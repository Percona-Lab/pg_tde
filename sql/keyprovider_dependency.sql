CREATE EXTENSION pg_tde;

SELECT pg_tde_add_key_provider_file('mk-file','/tmp/pg_tde_test_keyring.per');
SELECT pg_tde_add_key_provider_file('free-file','/tmp/pg_tde_test_keyring_2.per');
SELECT pg_tde_add_key_provider_vault_v2('V2-vault','vault-token','percona.com/vault-v2/percona','/mount/dev','ca-cert-auth');

SELECT pg_tde_set_master_key('test-db-master-key','mk-file');

-- Try dropping the in-use key provider
DELETE FROM percona_tde.pg_tde_key_provider WHERE provider_name = 'mk-file'; -- Should fail
-- Now delete the un-used  key provider
DELETE FROM percona_tde.pg_tde_key_provider WHERE provider_name = 'free-file'; -- Should pass
DELETE FROM percona_tde.pg_tde_key_provider WHERE provider_name = 'V2-vault'; -- Should pass

DROP EXTENSION pg_tde;
