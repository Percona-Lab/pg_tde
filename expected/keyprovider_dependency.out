CREATE EXTENSION pg_tde;
SELECT pg_tde_add_key_provider_file('mk-file','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_key_provider_file 
------------------------------
                            1
(1 row)

SELECT pg_tde_add_key_provider_file('free-file','/tmp/pg_tde_test_keyring_2.per');
 pg_tde_add_key_provider_file 
------------------------------
                            2
(1 row)

SELECT pg_tde_add_key_provider_vault_v2('V2-vault','vault-token','percona.com/vault-v2/percona','/mount/dev','ca-cert-auth');
 pg_tde_add_key_provider_vault_v2 
----------------------------------
                                3
(1 row)

SELECT pg_tde_set_master_key('test-db-master-key','mk-file');
 pg_tde_set_master_key 
-----------------------
 
(1 row)

-- Try dropping the in-use key provider
DELETE FROM percona_tde.pg_tde_key_provider WHERE provider_name = 'mk-file'; -- Should fail
ERROR:  Key provider "mk-file" cannot be deleted
DETAIL:  The master key for the database depends on this key provider.
-- Now delete the un-used  key provider
DELETE FROM percona_tde.pg_tde_key_provider WHERE provider_name = 'free-file'; -- Should pass
DELETE FROM percona_tde.pg_tde_key_provider WHERE provider_name = 'V2-vault'; -- Should pass
DROP EXTENSION pg_tde;
